### PR TITLE
Monitoring: Asset Deviation

### PR DESCRIPTION
**Pull request by Gauthamastro**
Closes https://github.com/Polkadex-Substrate/issue-tracker/issues/240

### CODE REVIEW

File: pallets/ocex/src/rpc.rs [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+	/// This is a blocking call for offchain worker.
+	pub fn calculate_inventory_deviation() -> Result<BTreeMap<AssetId, Decimal>, DispatchError> {
+		// Acquire the lock to run off-chain worker
+		if let Ok(_) = Self::acquire_offchain_lock() {
+			//      2. Load last processed blk
+			let mut root = crate::storage::load_trie_root();
+			log::info!(target:"ocex-rpc","state_root {:?}", root);
+			let mut storage = crate::storage::State;
+			let mut state = OffchainState::load(&mut storage, &mut root);
+			let state_info = Self::load_state_info(&mut state)?;
```
Comment (zktony) [inline pallets/ocex/src/rpc.rs]: What if it fails? Who is going to release the lock?

Comment (Gauthamastro) [reply pallets/ocex/src/rpc.rs]: Resolved by moving it out

File: pallets/ocex/src/rpc.rs
Comment (serhii-temchenko) [inline pallets/ocex/src/rpc.rs:1]: This module lacks documentation and license

File: pallets/ocex/src/rpc.rs [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+			}
+		}
+		Err(DispatchError::Other("Offchain storage mutex error"))
+	}
+
+	/// Release offchain storage lock
+	pub fn release_offchain_lock() {
+		// Check if another worker is already running or not
+		let s_info = StorageValueRef::persistent(&WORKER_STATUS);
+		s_info.set(&false); // Set WORKER_STATUS to true
```
Comment (serhii-temchenko) [inline pallets/ocex/src/rpc.rs:50]: Why comment says set to `true` and `s_info.set(&false)` set's false?

File: pallets/ocex/src/rpc.rs [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+	}
+
+	pub fn get_balances(
+		state: &mut OffchainState,
+		account: &AccountId,
+	) -> Result<BTreeMap<AssetId, Decimal>, &'static str> {
```
Comment (serhii-temchenko) [inline pallets/ocex/src/rpc.rs:56]: Please add docs

File: pallets/ocex/src/rpc.rs [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+		let mut storage = crate::storage::State;
+		let mut state = OffchainState::load(&mut storage, &mut root);
+		let state_info = Self::load_state_info(&mut state)?;
+		let last_processed_blk = state_info.last_block;
+		//      2. Load all main accounts and registered assets from on-chain
+		let mut offchain_inventory = BTreeMap::new();
+		for (main, _) in <Accounts<T>>::iter() {
+			//      3. Compute sum of all balances of all assets
+			let balances: BTreeMap<AssetId, Decimal> =
+				Self::get_balances(&mut state, &Decode::decode(&mut &main.encode()[..]).unwrap())?;
```
Comment (serhii-temchenko) [inline pallets/ocex/src/rpc.rs:80]: Shouldn't unwrap be replaced with proper result handling?

**Pull request merged**